src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -858,10 +850,29 @@ internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttrib
             var data = (CommonParameterWellKnownAttributeData)decodedData;
             if (data != null)
             {
-                if (this.RefKind == RefKind.Ref && data.HasOutAttribute && !data.HasInAttribute)
+                switch (RefKind)
                 {
-                    // error CS0662: '...' cannot specify only Out attribute on a ref parameter. Use both In and Out attributes, or neither.
-                    diagnostics.Add(ErrorCode.ERR_OutAttrOnRefParam, this.Locations[0]);
+                    case RefKind.Ref:
+                        if (data.HasOutAttribute && !data.HasInAttribute)
+                        {
+                            // error CS0662: Cannot specify only Out attribute on a ref parameter. Use both In and Out attributes, or neither.
```

Comment:
nit: This is not accurate error message since `[In] ref` or `[In, Out] ref` is allowed (both are not required).

This is completely separate from the issue you are fixing and is just an existing "inconsistency" with the reported error message.

Reply:
Thanks for the clarification. I was wondering what would happen with `[In] ref`... It's allowed.

---
In reply to: [143864191](https://github.com/dotnet/roslyn/pull/22645#discussion_r143864191) [](ancestors = 143864191)

Reply:
Good point. I'll fix it here since I'm modifying that area anyways.